File: combat.rb
#!/usr/bin/env ruby

class Weapon
    attr_accessor :type, :weight, :range, :armor_class

    def Weapon.set_attributes( tp="null", wt=0, rg=0, ac=0 )
        w = Weapon.new
        w.type        = tp
        w.weight      = wt
        w.range       = rg
        w.armor_class = ac
        return w
    end
end

tank1 = Weapon.set_attributes( "M1 Abrams", 15000, 130, 3  ) 
tank2 = Weapon.set_attributes( "Panzer",    16000,  80, 14 )
tank3 = Weapon.set_attributes( "Sherman",   10000,  90, 7  )

weapon_arr = [ tank1, tank2, tank3 ]
for tank in weapon_arr
    puts "#{ tank.type } #{ tank.weight } #{ tank.range } #{ tank.armor_class }"
end

armor_classes = [ tank1.armor_class, tank2.armor_class, tank3.armor_class ]
armor_classes.sort
winning_armor_class = armor_classes[0]
puts winning_armor_class

tanks = [ tank1, tank2, tank3 ]
for i in tanks
    if i.armor_class == winning_armor_class
        winning_tank = i.type
        puts winning_tank
    end
end

File: combat.js
function Weapon ( type ) {
    this.type        = type;
    this.weight      = 0;
    this.range       = 0;
    this.armor_class = 0;
}

Weapon.prototype.getWeaponInfo = function() {
        return this.type + ' ' + this.weight + ' ' + this.range + ' ' + this.armor_class;
};

var tank1 = new Weapon( 'Sherman' );
tank1.weight      = 10000;
tank1.range       = 130;
tank1.armor_class = 15;

var tank2 = new Weapon( 'M1 Abrams' );
tank2.weight      = 20000;
tank2.range       = 200;
tank2.armor_class = 3;

var tank3 = new Weapon( 'Tiger' ); 
tank3.weight      = 14000;
tank3.range       = 120;
tank3.armor_class = 7;

function getLowestArmorClass() {
    var tank_array = [ tank1.armor_class, tank2.armor_class, tank3.armor_class ];
    tank_array.sort( function( a, b ) {
        return a - b;
    });
    var lowest = tank_array[0];
    return lowest;
}

function getWinningTank() {
    var lowest = getLowestArmorClass();
    if ( lowest == tank1.armor_class ) {
        return tank1.type;
    } else if ( lowest == tank2.armor_class ) {
        return tank2.type;
    } else if ( lowest == tank3.armor_class ) {
        return tank3.type;
    }
}

document.writeln(); 
document.writeln( tank1.getWeaponInfo() ); 
document.writeln( tank2.getWeaponInfo() );
document.writeln( tank3.getWeaponInfo() ); 
document.writeln( 'the winning armor class is: ' + getLowestArmorClass() ); 
document.writeln( 'the winning tank is: ' + getWinningTank() ); 
